src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,61 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURES_NUMBER = 5;
+    private static final double DEFAULT_RADIUS = 10;
+    private static final Color CIRCLE_DEFAULT_COLOR = Color.WHITE;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureNumber = random.nextInt(FIGURES_NUMBER);
+        Color randomColor = colorSupplier.getRandomColor();
+        switch (figureNumber) {
+            case 0:
+                return getRandomCircle(randomColor);
+            case 1:
+                return getRandomRightTriangle(randomColor);
+            case 2:
+                return getRandomSquare(randomColor);
+            case 3:
+                return getRandomRectangle(randomColor);
+            default:
```

Comment:
Consider using the `FIGURES_NUMBER` constant instead of hardcoding the default case value. This will make the code more maintainable and easier to update if the number of figures changes.